File: fixtures/scale/java/medium/src/calc/funcs/PowerFunctions.java
package calc.funcs;

import calc.eval.FunctionRegistry;

public class PowerFunctions {
    public static void register(FunctionRegistry registry) {
        registry.addUnary("sqrt", Math::sqrt);
        registry.addUnary("cbrt", Math::cbrt);
        registry.addUnary("square", x -> x * x);
        registry.addAggregate("hypot", args -> Math.hypot(args[0], args[1]));
        registry.addAggregate("pow", args -> Math.pow(args[0], args[1]));
    }
}

File: fixtures/scale/swift/medium/Functions/PowerFunctions.swift
import Foundation

enum PowerFunctions {
    static func install(_ registry: FunctionRegistry) {
        registry.define("sqrt") { sqrt($0[0]) }
        registry.define("cbrt") { cbrt($0[0]) }
        registry.define("pow") { pow($0[0], $0[1]) }
        registry.define("hypot") { hypot($0[0], $0[1]) }
    }
}
